internal/versions/versions.go

```diff
@@ -103,6 +106,29 @@ func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params o
 			supportedArchs = []string{*releaseImage.CPUArchitecture}
 		}
 
+		// (MGMT-11859) We are filtering out multiarch release images so that they are available
+		//              only for customers allowed to use them. This is in order to be able to
+		//              expose them in OCP pre-4.13 without making them generally available.
+		if len(supportedArchs) > 1 {
+			hasMultiarchAuthorization := false
+			checkedForMultiarchAuthorization := false
+
+			if !checkedForMultiarchAuthorization {
```

Comment:
Please explain.
This if statement is always `true`, after which you change `checkedForMultiarchAuthorization` to `true`. Why?

Reply:
Yeah you are right, the initial value for the bools is now moved to the very top of this function to ensure that we do "check for authorization" only once in the whole call